src/test/app/Vault_test.cpp

```diff
@@ -2843,6 +3190,870 @@ class Vault_test : public beast::unit_test::suite
+                return {
+                    Account("vault", vault->at(sfAccount)),
+                    vault->at(sfShareMPTID)};
+            }(keylet);
+            MPTIssue shares(issuanceId);
+            env.memoize(vaultAccount);
+
+            auto const peek =
+                [=, &env, this](std::function<bool(SLE&, SLE&)> fn) -> bool {
+                return env.app().openLedger().modify(
+                    [&](OpenView& view, beast::Journal j) -> bool {
+                        Sandbox sb(&view, tapNONE);
+                        auto vault = sb.peek(keylet::vault(keylet.key));
+                        if (!BEAST_EXPECT(vault != nullptr))
+                            return false;
+                        auto shares = sb.peek(
+                            keylet::mptIssuance(vault->at(sfShareMPTID)));
+                        if (!BEAST_EXPECT(shares != nullptr))
+                            return false;
+                        if (fn(*vault, *shares))
+                        {
+                            sb.update(vault);
+                            sb.update(shares);
+                            sb.apply(view);
+                            return true;
+                        }
+                        return false;
+                    });
+            };
+
+            test(
+                env,
+                {.owner = owner,
+                 .issuer = issuer,
+                 .depositor = depositor,
+                 .vaultAccount = vaultAccount,
+                 .shares = shares,
+                 .share = PrettyAsset(shares),
+                 .vault = vault,
+                 .keylet = keylet,
+                 .assets = asset.raw().get<Issue>(),
+                 .asset = asset,
+                 .peek = peek});
+        };
+
+        testCase(18, [&, this](Env& env, Data d) {
+            testcase("Scale deposit overflow on first deposit");
+            auto tx = d.vault.deposit(
+                {.depositor = d.depositor,
+                 .id = d.keylet.key,
+                 .amount = d.asset(10)});
+            env(tx, ter{tecPATH_DRY});
+            env.close();
+        });
+
+        testCase(18, [&, this](Env& env, Data d) {
+            testcase("Scale deposit overflow on second deposit");
+
+            {
+                auto tx = d.vault.deposit(
+                    {.depositor = d.depositor,
+                     .id = d.keylet.key,
+                     .amount = d.asset(5)});
+                env(tx);
+                env.close();
+            }
+
+            {
+                auto tx = d.vault.deposit(
+                    {.depositor = d.depositor,
+                     .id = d.keylet.key,
+                     .amount = d.asset(10)});
+                env(tx, ter{tecPATH_DRY});
+                env.close();
+            }
+        });
+
+        testCase(18, [&, this](Env& env, Data d) {
+            testcase("Scale deposit overflow on total shares");
```

Comment:
I think this test case is identical to the previous one, `"Scale deposit overflow on second deposit"`

Reply:
it's not. The 5 being added here does not result in an exception (thrown, then captured and converted into `tecPATH_DRY`) like in above test. Instead we go one step further to attempt to create new shares for the depositor, and receive an error inside `accountSend` from `maximumAmount` check inside `rippleSendMPT` at https://github.com/XRPLF/rippled/blob/d145666198610d73e32951d9b9039c3ecb7caa92/src/xrpld/ledger/detail/View.cpp#L2056

Reply:
> it's not. The 5 being deposited here does not result in an exception (thrown, then captured and converted into `tecPATH_DRY`) like in the above test, where the amount was 10. Instead we go one step further to attempt to create new shares for the depositor, and receive an error inside `accountSend` from `maximumAmount` check inside `rippleSendMPT` at

Oh! Ok, I missed that the second deposit was 10 in the first test case and 5 in the second. Thanks.